pycparser/c_ast.py

```diff
@@ -96,14 +101,23 @@ def show(self, buf=sys.stdout, offset=0, attrnames=False, nodenames=False, showc
             buf.write(' (at %s)' % self.coord)
         buf.write('\n')
 
+        indent += "  " if islast else "│ "
+        
+        lastChild = None
+        
+        children = self.children()
+        if len(children) > 0:
+            _, lastChild = children[-1]
```

Comment:
I think you should be able to accomplish this in a cleaner way using `enumerate` in the `for` loop below. Then you can trivially find if the index is the one before last and you won't need this condition and `lastChild`

Reply:
I think I understood what you meant but if not let me know